### DISCUSSION TITLE
Apollo Federation (RoverCli)

### DISCUSSION THREAD

**Discussion by elritz** [hayes/pothos#744] [Q&A]
Hi, Im using two server with pothos on both one called Profiling one called Managing. Im trying to be able to return from Managing a connection to Profiling on the Profile Type..
So what i have done in Profiling is:
```
export const Profile = builder.prismaObject("Profile", {
  fields: (t) => ({
    id: t.exposeString("id"),
    createdAt: t.expose("createdAt", {
      type: "DateTime",
    }),
    updatedAt: t.expose("updatedAt", {
      type: "DateTime",
    }),
  }),
});


builder.asEntity(Profile, {
  key: builder.selection<{ id: string }>("id"),
  resolveReference: async (user) => {
    return prismaClient.profile.findFirst({ where: { id: user.id } });
  },
});
```

I am able to move forward with the rover-cli 

Now on the Managing Service side I want to be able to return the Profile Object I have here right. I would say this is pretty standard expectation

Here is what I have on that 
```
export const Profile = builder
  .externalRef("Profile", builder.selection<{ id: string }>("id"))
  .implement({
    externalFields: (t) => ({
      id: t.string({ nullable: true }),
    }),
  });
```

But the error that I get on the the rover cli is 

```
WARN: The following build errors occurred:
EXTENSION_WITH_NO_BASE: [profiling] Type "Profile" is an extension type, but there is no type definition for "Profile" in any subgraph.
```

When I run an introspection on my endpoints locally, my super graph schema for the Profile type is as follows

```
type Profile
  @join__type(graph: MANAGING, key: "id", extension: true)
  @join__type(graph: PROFILING, key: "id")
{
  id: String @join__field(graph: MANAGING, type: "String") @join__field(graph: PROFILING, type: "String!")
  createdAt: DateTime! @join__field(graph: PROFILING)
  updatedAt: DateTime! @join__field(graph: PROFILING)
}
```

Does this mean that a Q/M that goes that runs through the Managing service is able to return values based on the type Profile (which came from Profiling). 
Am I building these Object entities correctly from where type ownership graph to the subgraphs that need it.  
Also the code in the docs uses objectRef, 

const UserType = builder.objectRef<User>('User').implement({
  fields: (t) => ({
    id: t.exposeID('id'),
    name: t.exposeString('name'),
    username: t.exposeString('username'),
  }),
});
 
I am using prismaObject (I’ve also tried objectRef) What is the difference and if there is a way to utilize prisma on this route 

Thanks

**Comment by hayes** [comment]
This all looks correct to me, and I don't think there should be issues using a prismaObject instead of an objectRef.

I wonder if this warning happens because Pothos adds the `@extends` directive to all external refs, even if they do not add any additional fields.  Something worth checking would be to add a field (can be anything, eg `t.string({ resolve: () => 'test' })`) and see if that causes the warning to go away.   I think federation v1 required more extends, and I originally wrote the federation plugin when federation docs were fairly incomplete.

If adding a field to the subgraph that defines the externalRef makes the warning go away, the above is likely the issue, and we just need to update the plugin to only add the extends directive if the type actually adds fields.

**Reply by elritz** [reply]
Thank you for the reply and with the explanation as well , I will keep that in mind if i have a similar issue. My problem was unrelated to Pothos.

**Reply by hayes** [reply]
cool, glad you got it figured out!
